lib/config.c

```diff
@@ -428,14 +429,55 @@ walk_host_config(
                 log_internal(MTC_LOG_ERR, "%s: failed to get IPaddress\n", __func__);
                 return MTC_ERROR_CF_INVALID_FORMAT;
             }
-            c->common.host[c->common.hostnum].ip_address = inet_addr((char *) txt);
-            if (c->common.host[c->common.hostnum].ip_address == 0) 
+
+            const char *ip_address = (char *) txt;
+
+            struct addrinfo hints;
+            struct addrinfo *result;
+            memset(&hints, 0, sizeof hints);
+            hints.ai_family = AF_UNSPEC;
+            hints.ai_socktype = SOCK_DGRAM;
+            hints.ai_flags = AI_NUMERICHOST;
+            hints.ai_protocol = 0;
+            hints.ai_canonname = NULL;
+            hints.ai_addr = NULL;
+            hints.ai_next = NULL;
+
+            errno = 0;
+            const int ret = getaddrinfo(ip_address, NULL, &hints, &result);
+            if (ret)
             {
-                log_internal(MTC_LOG_ERR, "%s: invalid IPaddress %s\n", __func__, txt);
+                log_internal(MTC_LOG_ERR, "%s: failed to get addr info of `%s` (info: %s). (sys %d)\n",
+                    __func__, ip_address, gai_strerror(ret), errno);
                 xmlFree(txt);
+                freeaddrinfo(result);
```

Comment:
If `getaddrinfo()` fails, result wouldn't be set to anything so there is no need to free here. Since result is uninitialized, it might crash (getaddrinfo() may set result to NULL but we shouldn't rely on that).